### PR TITLE
Fixes #16280: Directives appears both skipped and enforce

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/16280

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala [hunk 1]
```diff
@@ -58,6 +58,21 @@ import com.normation.rudder.reports.ReportsDisabled
...
 import com.normation.utils.Control.sequence
 
+
+object ReportingServiceUtils {
+
+  /*
+   * Build rule status reports from node reports, decide=ing which directive should be "skipped"
+   */
+  def buildRuleStatusReport(ruleId: RuleId, nodeReports: Map[NodeId, NodeStatusReport]): RuleStatusReport = {
+    val toKeep = nodeReports.values.flatMap( _.report.reports ).toList
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala]: here, you should probaly add a filter on the ruleId, to ensure that nodestatusreports don't have extra rules that would polute it

Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala]:
```suggestion
    val toKeep = nodeReports.values.flatMap( _.report.reports ).filter(_.ruleId==ruleId).toList
```

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**